src/Sk/SmartId/Util/Curl.php

```diff
@@ -325,7 +333,29 @@ public function getError()
     return false;
   }
 
-  /**
+  public static function useOnlyDemoPublicKey()
+  {
+      self::$server_ssl_public_keys = self::DEMO_SID_PUBLIC_KEY;
+  }
+
+  public static function useOnlyLivePublicKey()
+  {
+      self::$server_ssl_public_keys =
+              self::RP_API_PUBLIC_KEY_VALID_FROM_2016_12_20_TO_2020_01_19 . ";"
+              . self::RP_API_PUBLIC_KEY_VALID_FROM_2019_11_01_TO_2021_11_05;
+  }
+
+    public static function setPublicKeysFromArray(array $public_keys)
```

Comment:
Good to see someone is solving SSL pinning issue. Doesn't this approach violate encapsulation? Wouldn't it be more wise to add extra layer so the pinning can be configured through _AuthenticationRequestBuilder_?

Reply:
Hmm, shouldn't the pinning be responsibility of _AuthenticationResponseValidator_?  The downside is that in this way it is not possible to use pinning functionality provided by curl.

Reply:
> Hmm, shouldn't the pinning be responsibility of _AuthenticationResponseValidator_?

Probably not. Digging in code reveals that this is meant to validate data about authenticated user. After the successful authentication server returns user data, and the certificate in it is actually user certificate. So _AuthenticationResponseValidator_ does not have no  idea about server certificate therefore can not validate

Reply:
> @andrevka could you change Curl::useOnlyDemoPublicKey() and others to be non-static?
Did it

Reply:
> > Hmm, shouldn't the pinning be responsibility of _AuthenticationResponseValidator_?
> 
> Probably not. Digging in code reveals that this is meant to validate data about authenticated user. After the successful authentication server returns user data which is the input of _\Sk\SmartId\Api\AuthenticationResponseValidator::validate_. The certificate in this input parameter is actually user certificate. Moreover, _AuthenticationResponseValidator_ does not have information about server certificate/public key therefore can not carry out this responsibility anyway.

I think i have resolved the issue, you can check it out.